src/main/scala/io/iohk/ethereum/mallet/service/RpcClient.scala

```diff
@@ -0,0 +1,108 @@
+package io.iohk.ethereum.mallet.service
+
+
+import java.util.UUID
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model._
+import akka.stream.ActorMaterializer
+import akka.util.ByteString
+import com.typesafe.config.ConfigFactory
+import io.circe.generic.auto._
+import io.circe.parser._
+import io.circe.syntax._
+import io.circe.{Decoder, Json}
+import io.iohk.ethereum.domain.Address
+import io.iohk.ethereum.jsonrpc.{JsonRpcError, TransactionReceiptResponse}
+import io.iohk.ethereum.mallet.common.{Constants, Err, RpcClientError, Util}
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+
+object RpcClient {
+
+  /**
+    * This factory method is defining an ActorSystem, ActorMaterializer and ExecutionContext for
+    * the [[RpcClient]]. To customize these dependencies use [[RpcClient]]'s constructor
+    */
+  def apply(node: Uri): RpcClient = {
```

Comment:
@LukasGasior1 hopefully this should address your remarks about reusability, while limiting the scope of dependencies in mallet.

Reply:
ok, let's leave it that way